inception-recommendation-api/src/main/java/de/tudarmstadt/ukp/inception/recommendation/api/model/Predictions.java

```diff
@@ -207,6 +208,19 @@ public AnnotationObject getPredictionByVID(VID aVID)
                 .collect(Collectors.toList()).get(0);
     }
 
+    /**
+     * Returns the first prediction that matches recommendationId and recommenderId
+     * in the given document
+     */
+    public AnnotationObject getPredictionInDocumentByVID(SourceDocument document, VID aVID)
+    {
+        return predictions.values().stream()
+                .filter(f -> f.getDocumentName().equals(document.getName()))
+                .filter(f -> f.getId() == aVID.getSubId())
+                .filter(f -> f.getRecommenderId() == aVID.getId())
+                .collect(Collectors.toList()).get(0);
```

Comment:
We do not need to collect the full list then to only return the first element, right? That seems inefficient. Also, we get a nasty exception in case no annotation with the given document/VID is found. Either return null or maybe an optional.